library/ssl_tls13_generic.c

```diff
@@ -1386,9 +1386,9 @@ int mbedtls_ssl_reset_transcript_for_hrr( mbedtls_ssl_context *ssl )
 
     hash_len += 4;
 
+#if defined(MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA)
```

Comment:
Note: here this is shared with TLS 1.2, so the `HAS_ALG_..._BASED_ON_USE_PSA` macro is indeed the correct choice.